src/catalogue/cmd/cataloguesvc/main.go

```diff
@@ -109,19 +110,37 @@ func main() {
 	}
 
 	// Data domain.
+	/ Try connecting to the primary database
 	db, err := sqlx.Open("godror", *connectString)
 	if err != nil {
-		logger.Log("err", err)
-		os.Exit(1)
+		logger.Log("Error", "Failed to open Primary Database connection", "CONNECTSTRING", *connectString)
 	}
-	defer db.Close()
 
-	// Check if DB connection can be made, only for logging purposes, should not fail/exit
+	// Test the connection to the primary database
 	err = db.Ping()
 	if err != nil {
-		logger.Log("Error", "Unable to connect to Database", "CONNECTSTRING", connectString)
+		logger.Log("Error", "Unable to connect to Primary Database", "CONNECTSTRING", *connectString)
```

Comment:
Same here. Also, the log should show that tried to the main DB and then is trying the standby.